test/org/sagebionetworks/bridge/services/StudyServiceMockTest.java

```diff
@@ -121,6 +122,59 @@ private void assertDirectoryUpdated(Consumer<Study> consumer) {
         verify(directoryDao).updateDirectoryForStudy(study);
     }
 
+    @Test(expected = EntityNotFoundException.class)
+    public void deactivateStudyAlreadyDeactivatedBefore() {
+        Study study = getTestStudy();
+        study.setActive(false);
+        when(studyDao.getStudy(study.getIdentifier())).thenReturn(study);
+
+        service.deleteStudy(study.getIdentifier(), false);
+    }
+
+    @Test(expected = EntityNotFoundException.class)
+    public void deactivateStudyNotFound() {
+        Study study = getTestStudy();
+        study.setActive(false);
+        when(studyDao.getStudy(study.getIdentifier())).thenReturn(null);
```

Comment:
Does this method return null, rather than throwing EntityNotFoundException? Just checking. Also I would verify that the dao is not called to delete the study.